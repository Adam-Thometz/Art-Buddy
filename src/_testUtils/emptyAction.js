const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION