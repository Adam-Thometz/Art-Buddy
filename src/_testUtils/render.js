import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
