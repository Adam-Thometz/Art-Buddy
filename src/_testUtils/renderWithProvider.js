import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;