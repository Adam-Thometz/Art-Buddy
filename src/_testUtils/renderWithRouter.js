import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;