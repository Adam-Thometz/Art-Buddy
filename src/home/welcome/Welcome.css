.Welcome {
  display: flex;
  flex-direction: column;
}

.Welcome-description {
  text-align: left;
  width: 90%;
  margin: 0 auto;
}

.Welcome button {
  width: 50%;
  margin: 0 auto;
}
