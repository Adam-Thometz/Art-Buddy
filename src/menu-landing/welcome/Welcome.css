.Welcome-description {
  text-align: left;
  width: 90%;
  margin: 0 auto;
}