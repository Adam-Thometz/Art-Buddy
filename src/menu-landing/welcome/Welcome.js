import React from "react";

import './Welcome.css';

import WindowNavbar from "../../_components/window-nav/WindowNavbar";
import Button from "../../_components/button/Button";

const Welcome = ({ setHasVisited }) => {
  const handleVisited = () => {
    setHasVisited(true);
  };
  
  return (
    <section className="Welcome">
      <WindowNavbar page='WELCOME TO ART BUDDY!' />
      <article className="Welcome-description">
        <p>ARTBUDDY is a collection of fun and interactive fine arts games designed specifically with special education goals in mind</p>
        <p>Click below to get started</p>
      </article>
      <Button colorId={0} onClick={handleVisited}>LET'S GO!</Button>
    </section>
  );
};

export default Welcome;