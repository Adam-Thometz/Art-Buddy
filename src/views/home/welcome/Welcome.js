import "./Welcome.css";

import WindowNavbar from "components/window-nav/WindowNavbar";
import Button from "components/button/Button";

const Welcome = ({ setHasVisited }) => {
  const handleVisited = () => setHasVisited(true);

  return (
    <section className="Welcome">
      <WindowNavbar page="WELCOME TO ART BUDDY!" />
      <article className="Welcome-description">
        <p>
          ART BUDDY is a collection of fun and interactive fine arts games
          designed specifically with academic goals in mind
        </p>
        <p>Click below to get started</p>
      </article>
      <Button colorId={0} onClick={handleVisited}>
        LET'S GO!
      </Button>
    </section>
  );
};

export default Welcome;
