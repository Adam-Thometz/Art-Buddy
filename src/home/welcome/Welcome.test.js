import Welcome from "./Welcome";

import { render } from "_testUtils/render";

describe("Welcome component", () => {
  it("renders without crashing", () => {
    render(<Welcome />);
  });

  it("matches the snapshot of the welcome page", () => {
    const { asFragment } = render(<Welcome />);
    expect(asFragment()).toMatchSnapshot();
  });
});
