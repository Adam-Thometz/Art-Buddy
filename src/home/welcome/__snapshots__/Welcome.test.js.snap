// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component matches the snapshot of the welcome page 1`] = `
<DocumentFragment>
  <section
    class="Welcome"
  >
    <header
      class="WindowNavbar-games"
    >
      <h1
        class="WindowNavbar-label"
      >
        WELCOME TO ART BUDDY!
      </h1>
    </header>
    <article
      class="Welcome-description"
    >
      <p>
        ART BUDDY is a collection of fun and interactive fine arts games designed specifically with academic goals in mind
      </p>
      <p>
        Click below to get started
      </p>
    </article>
    <button
      class="Button regular"
      style="border-color: #89eb0a;"
    >
      LET'S GO!
    </button>
  </section>
</DocumentFragment>
`;
